openfisca_aotearoa/variables/regulation/student_allowance/interpretation.py

```diff
@@ -0,0 +1,122 @@
+
+
+class student_allowance__has_a_supported_child(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = MONTH
+    label = "has a supported child as per Student Allowances Regulations 1998"
+    reference = """
+    https://legislation.govt.nz/regulation/public/1998/0277/latest/whole.html#DLM259968
+    supported child, in relation to a student applying for an allowance, means a person younger than 24—
+        (a)
+
+        whose well-being and financial support are the responsibility of the student; and
+        (b) who lives with that student at least half of the time; and
+        (c) who is not in receipt of—
+            (i) an allowance continued by regulation 3(a) to (c); or
+            (ii) jobseeker support, sole parent support, an emergency benefit, or a supported living payment under the Social Security Act 1964; or
+            (iii) New Zealand superannuation under the New Zealand Superannuation and Retirement Income Act 2001 or a veteran’s pension under
+                the Veterans’ Support Act 2014; or
+            (iv) payments under any government-assisted scheme (other than the Student Loan Scheme) which, in the opinion of the chief executive,
+                is similar to a benefit under the Social Security Act 1964; or
+            (v) income before tax from employment or self-employment which exceeds $80 per week; and
+        (d) in respect of whom no orphan’s benefit or unsupported child’s benefit is payable under the Social Security Act 1964
+    """
+
+
+class student_allowance__partner_has_a_supported_child(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = MONTH
+    label = "their spouse has a supported child, as per Student Allowances Regulations 1998"
+
+
+class student_allowance__has_a_spouse(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = MONTH
+    label = "Has spouse as per Student Allowances Regulations 1998"
+    reference = """
+    https://legislation.govt.nz/regulation/public/1998/0277/latest/whole.html#DLM259958
+    Spouse, in relation to an applicant, means a person who is legally married to that applicant if—
+        (a) both of them are of or over 24; or
+        (b) one or both of them are younger than 24 and at least 1 of them has a supported child"""
+
+    def formula(persons, period, parameters):
+        part_a = (persons('age', period) >= 24) * (persons('age_of_partner', period) >= 24)
+        part_b = ((persons('age', period) >= 24) + (persons('age_of_partner', period) >= 24)) * \
+            (persons('student_allowance__has_a_supported_child', period) + persons('student_allowance__partner_has_a_supported_child', period))
+
+        return part_a + part_b
+
+
+class student_allowance__is_a_student(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = MONTH
+    label = "student means a person who is enrolled or intends to enrol in a recognised course of study"
```

Comment:
Needs a reference to: http://www.legislation.govt.nz/regulation/public/1998/0277/latest/whole.html#DLM259966